APMrover2/GCS_Rover.h

```diff
@@ -9,24 +9,21 @@ class GCS_Rover : public GCS
 
 public:
 
-    // return the number of valid GCS objects
-    uint8_t num_gcs() const override { return ARRAY_SIZE(_chan); };
-
     // return GCS link at offset ofs
-    GCS_MAVLINK_Rover &chan(uint8_t ofs) override {
-        if (ofs >= num_gcs()) {
+    GCS_MAVLINK_Rover *chan(const uint8_t ofs) override {
+        if (ofs > _num_gcs) {
             AP::internalerror().error(AP_InternalError::error_t::gcs_offset);
-            ofs = 0;
+            return nullptr;
```

Comment:
@OXINARF Should we change this to return chan[0] as we were?  We don't need to change this behaviour as part of this PR.

Reply:
I don't think we should, channel 0 might not exist either. 

I think we just need to fix one case where we assume that at least channel 0 exists (https://github.com/ArduPilot/ardupilot/pull/11617/files/ff81a5248f1e4c4ec73f5b19d2ed660d16aa1a06#diff-184f24112b8a738e36349802bd5628eeR345) and then we are safe.

Reply:
Fixed.